tests/tracking/_tracking_service/test_utils.py

```diff
@@ -196,6 +196,46 @@ def test_get_store_databricks_profile():
         assert "mycoolprofile" in str(e_info.value)
 
 
+def test_get_store_caches_on_store_uri_and_artifact_uri(tmpdir):
+    registry = mlflow.tracking._tracking_service.utils._tracking_store_registry
+
+    store_uri_1 = "sqlite:///backend_store_1.db"
+    store_uri_2 = os.path.join(str(tmpdir), "backend_store_2")
```

Comment:
Nit: Just feel free to ignore this comment :) I just wanted to share a simpler way (that I learned one month ago) to join file paths.

```suggestion
    store_uri_2 = tmpdir.join("backend_store_2").strpath
    # or
    # store_uri_2 = str(tmpdir.join("backend_store_2"))
```

Reply:
Great suggestion! I've adopted this throughout test cases added in this PR.